src/apps/Agreement/VersionHistory.js

```diff
@@ -0,0 +1,101 @@
+import React, { useMemo } from 'react'
+import {
+  GU,
+  useTheme,
+  ContextMenu,
+  ContextMenuItem,
+  IconView,
+  useLayout,
+} from '@aragon/ui'
+import PropTypes from 'prop-types'
+
+function VersionHistory({ items }) {
+  const itemsToRender = useMemo(
+    () =>
+      items.map((item, i) => {
+        return {
+          title:
+            i === items.length - 1 ? 'Created Agreement' : 'Updated Agreement',
+          date: item,
+        }
+      }),
+    [items]
+  )
+
+  return (
+    <React.Fragment>
+      {itemsToRender.map(({ title, date }, i) => (
+        <HistoryEntry title={title} date={date} key={i} />
+      ))}
+    </React.Fragment>
+  )
+}
+
+/* eslint-disable react/prop-types */
+function HistoryEntry({ title, date }) {
+  const theme = useTheme()
+  const { layoutName } = useLayout()
+
+  return (
+    <div
+      css={`
+        display: flex;
+        justify-content: space-between;
+        padding: ${layoutName === 'small' ? 2 * GU : 3 * GU}px;
+        & + & {
+          border-top: 1px solid ${theme.border};
+        }
+      `}
+    >
+      <div>
+        <h2
+          css={`
+            line-height: 1.1;
+            margin-bottom: ${1 * GU}px;
+          `}
+        >
+          {title}
+        </h2>
+        <p
+          css={`
+            line-height: 1.1;
+            color: ${theme.surfaceContentSecondary};
+          `}
+        >
+          {date}
+        </p>
+      </div>
+      <div>
```

Comment:
Just curious, do we need to set a `flex-shrink: 0` here to avoid this getting squeezed out if the title's length gets too long?

Reply:
Thanks for checking! the context menu already comes with a hard width so it'll break on word just fine.